app/Blocks/Accordion/src/edit.js

```diff
@@ -0,0 +1,40 @@
+import { __ } from '@wordpress/i18n';
+import { useBlockProps, useInnerBlocksProps } from '@wordpress/block-editor';
+
+export default function Edit( { attributes, setAttributes } ) {
+
+	const { showAll } = attributes;
+
+	const blockProps = useBlockProps(
+		{
+			className:"govuk-accordion",
+			id:"accordion-default"
+		}
+	);
+
+	const innerBlocksProps = useInnerBlocksProps( blockProps, {
+		defaultBlock: 'govukblogs/accordion-row',
+		template: [ [ 'govukblogs/accordion-row' ] ],
+		templateInsertUpdatesSelection: true,
+		allowedBlocks: ['govukblogs/accordion-row'] 
+	} );
+
+	const toggleAll = () => {
+		setAttributes({showAll: !showAll})
+	}
+
+	return (
+		<div className='govuk-frontend-supported'>
+			<div className="govuk-accordion__controls">
+				<button type="button" className="govuk-accordion__show-all" aria-expanded="false" onClick={toggleAll}>
+					<span className="govuk-accordion-nav__chevron govuk-accordion-nav__chevron--down"></span>
+					<span className="govuk-accordion__show-all-text">{showAll ? 'Hide all sections' : 'Show all sections'}</span>
```

Comment:
**question:** Is it possible to mirror the front-end functionality with this button i.e. when all accordion items have been opened individually, opening the last item triggers a switch to 'Hide all sections'? Currently, the button would still display and act as a 'Show all' so would need double-clicking to hide all items.

Reply:
I'm not sure - I think it would require quite a bit of complexity. Because in the block editor these are separate blocks each child block would need to pass props back up to the parent (which I'm not sure is possible using block context, which operates the other way round). The parent block would need to store this data and compare it to the number of child blocks.

Reply:
Ok, let's leave this as an open issue, for now.